tools/engine_tool/lib/src/dependencies.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:io' as io;
+
+import 'package:process_runner/process_runner.dart';
+
+import 'environment.dart';
+import 'logger.dart';
+
+/// Returns `false` if the Flutter engine's dependencies are up-to-date.
+Future<bool> dependenciesChanged() {
+  // TODO: This assumes dependencies never change.
+  // This should return `true` if the `DEPS` file changed since the last fetch.
+  return Future<bool>.value(false);
+}
+
+/// Update Flutter engine dependencies. Returns an exit code.
+Future<int> fetchDependencies(
+  Environment environment, {
+  bool verbose = false,
+}) async {
+  if (!environment.processRunner.processManager.canRun('gclient')) {
+    environment.logger.error('Cannot find the gclient command in your path');
+    return 1;
+  }
+
+  environment.logger.status('Fetching dependencies... ', newline: verbose);
+
+  Spinner? spinner;
+  if (!verbose) {
+    spinner = environment.logger.startSpinner();
+  }
+
+  final ProcessRunnerResult result = await environment.processRunner.runProcess(
+    <String>[
+      'gclient',
+      'sync',
+      '-D',
+    ],
+    runInShell: true,
+    startMode: verbose
+      ? io.ProcessStartMode.inheritStdio
+      : io.ProcessStartMode.normal,
+  );
+  spinner?.finish();
+  if (result.exitCode != 0) {
+    environment.logger.error('Fetching dependencies failed.');
```

Comment:
It'll probably be useful to emit the output from `gclient` when there's an error.

Reply:
Good idea, added!